
# GITHUB ISSUES REPORT FOR microsoft/debugpy


Generated on 2024-07-30 using: stale=30, all=False


Only showing items that are new to report in past 1 day(s)


---

## FOR ISSUES THAT ARE NOT MARKED AS BUGS:


### Issues in debugpy that need a response from team:

| Days Ago | Issue | Title |
| --- | --- | --- |
 | \* OP:0  |[1642](https://github.com/microsoft/debugpy/issues/1642 "Unable to set new variables or update exiting variables in debug console")  |Unable to set new variables or update exiting variables in debug console |

### Issues in debugpy that have comments from 3rd party after last team response:

| Days Ago | Issue | Title |
| --- | --- | --- |
 | \* P:1,  |[1252](https://github.com/microsoft/debugpy/issues/1252 "[Feature request] Support connect to ipv6 address")  |[Feature request] Support connect to ipv6 address |
 | \* P:0,  |[1636](https://github.com/microsoft/debugpy/issues/1636 "NameError: name 'x' not defined even though it should be")  |NameError: name 'x' not defined even though it should be |



### Issues with term '[sys.monitoring]'

| Days Ago | Issue | Title |
| --- | --- | --- |
 |  OP:257  |[1420](/issues/1420 "[sys.monitoring] Tracing execution flow") | [sys.monitoring] Tracing execution flow |
 |  OP:257  |[1421](/issues/1421 "[sys.monitoring] Stack tracing") | [sys.monitoring] Stack tracing |
 |  OP:257  |[1424](/issues/1424 "[sys.monitoring] Stepping") | [sys.monitoring] Stepping |
 |  OP:257  |[1428](/issues/1428 "[sys.monitoring] Stop on entry") | [sys.monitoring] Stop on entry |
 |  OP:257  |[1429](/issues/1429 "[sys.monitoring] Breakpoints") | [sys.monitoring] Breakpoints |
 |  OP:257  |[1433](/issues/1433 "[sys.monitoring] Breakpoint verification") | [sys.monitoring] Breakpoint verification |
 |  OP:257  |[1434](/issues/1434 "[sys.monitoring] Snap breakpoints to valid locations") | [sys.monitoring] Snap breakpoints to valid locations |
 |  OP:257  |[1435](/issues/1435 "[sys.monitoring] breakpoint() hook") | [sys.monitoring] breakpoint() hook |
 |  OP:256  |[1436](/issues/1436 "[sys.monitoring] Exception breakpoints") | [sys.monitoring] Exception breakpoints |
 |  OP:256  |[1437](/issues/1437 "[sys.monitoring] Expression evaluation") | [sys.monitoring] Expression evaluation |
 |  OP:256  |[1439](/issues/1439 "[sys.monitoring] Variable children") | [sys.monitoring] Variable children |
 |  OP:256  |[1440](/issues/1440 "[sys.monitoring] Variable values") | [sys.monitoring] Variable values |
 |  OP:256  |[1446](/issues/1446 "[sys.monitoring] Returned values") | [sys.monitoring] Returned values |
 |  OP:256  |[1447](/issues/1447 "[sys.monitoring] safe repr - strings") | [sys.monitoring] safe repr - strings |
 |  OP:256  |[1448](/issues/1448 "[sys.monitoring] safe repr - collections") | [sys.monitoring] safe repr - collections |
 |  OP:256  |[1449](/issues/1449 "[sys.monitoring] safe repr - numpy") | [sys.monitoring] safe repr - numpy |
 |  OP:256  |[1450](/issues/1450 "[sys.monitoring] safe repr - pandas") | [sys.monitoring] safe repr - pandas |
 |  OP:256  |[1451](/issues/1451 "[sys.monitoring] safe repr - tensorflow") | [sys.monitoring] safe repr - tensorflow |
 |  OP:256  |[1454](/issues/1454 "[sys.monitoring] Break on user-uncaught exception") | [sys.monitoring] Break on user-uncaught exception |
 |  OP:256  |[1455](/issues/1455 "[sys.monitoring] SystemExit handling") | [sys.monitoring] SystemExit handling |
 |  OP:256  |[1456](/issues/1456 "[sys.monitoring] Children - object attributes") | [sys.monitoring] Children - object attributes |
 |  OP:256  |[1457](/issues/1457 "[sys.monitoring] Children - sequence & set items") | [sys.monitoring] Children - sequence & set items |
 |  OP:256  |[1458](/issues/1458 "[sys.monitoring] Children - mapping items") | [sys.monitoring] Children - mapping items |
 |  OP:256  |[1459](/issues/1459 "[sys.monitoring] Children - numpy collections") | [sys.monitoring] Children - numpy collections |
 |  OP:256  |[1460](/issues/1460 "[sys.monitoring] Children - pandas collections") | [sys.monitoring] Children - pandas collections |
 |  OP:256  |[1461](/issues/1461 "[sys.monitoring] Children - tensorflow collections") | [sys.monitoring] Children - tensorflow collections |
 |  OP:251  |[1466](/issues/1466 "[sys.monitoring] Django breakpoints") | [sys.monitoring] Django breakpoints |
 |  OP:251  |[1467](/issues/1467 "[sys.monitoring] Flask (Jinja) breakpoints") | [sys.monitoring] Flask (Jinja) breakpoints |
 |  OP:251  |[1468](/issues/1468 "[sys.monitoring] Django template stack frames") | [sys.monitoring] Django template stack frames |
 |  OP:251  |[1469](/issues/1469 "[sys.monitoring] Jinja template stack frames") | [sys.monitoring] Jinja template stack frames |
 |  OP:251  |[1470](/issues/1470 "[sys.monitoring] Stdout redirection") | [sys.monitoring] Stdout redirection |
 |  OP:251  |[1471](/issues/1471 "[sys.monitoring] Performance telemetry") | [sys.monitoring] Performance telemetry |
 |  OP:251  |[1472](/issues/1472 "[sys.monitoring] Report loaded modules") | [sys.monitoring] Report loaded modules |
 |  OP:181  |[1507](/issues/1507 "[sys.monitoring] API to hide threads and files from debugger") | [sys.monitoring] API to hide threads and files from debugger |
 |  OP:181  |[1508](/issues/1508 "[sys.monitoring] Async stepping") | [sys.monitoring] Async stepping |
 |  OP:181  |[1509](/issues/1509 "[sys.monitoring] Just my code") | [sys.monitoring] Just my code |
 |  OP:125  |[1548](/issues/1548 "[sys.monitoring] Local variable cleanup is delayed by debugger") | [sys.monitoring] Local variable cleanup is delayed by debugger |




### Issues with term 'debugger'

| Days Ago | Issue | Title |
| --- | --- | --- |
 |  OP:1786  |[176](/issues/176 "Sequence diagrams and architecture diagrams for the debugger") | Sequence diagrams and architecture diagrams for the debugger |
 |  OP:1063  |[712](/issues/712 "notification like &quot;Failed launch debugger for child process xxxx&quot;.") | notification like "Failed launch debugger for child process xxxx". |
 |  OP:711  |[1155](/issues/1155 "Python debugger breaks on caught exception within a decorator and context manager") | Python debugger breaks on caught exception within a decorator and context manager |
 |  OP:633  |[1114](/issues/1114 "Display Python asyncio Tasks in VS Code Debugger") | Display Python asyncio Tasks in VS Code Debugger |
 |  OP:501  |[1245](/issues/1245 "Don't collapse nested lists/tuples into ... in debugger | make debugging more like PyCharm") | Don't collapse nested lists/tuples into ... in debugger | make debugging more like PyCharm |
 |  OP:474  |[1271](/issues/1271 "Timeout while using vscode debugger with any program which spawns a process") | Timeout while using vscode debugger with any program which spawns a process |
 |  OP:474  |[1272](/issues/1272 "When using debugger in remote environment it hangs and never successfully breaks when reaching a breakpoint") | When using debugger in remote environment it hangs and never successfully breaks when reaching a breakpoint |
 |  OP:460  |[1283](/issues/1283 "Python Debugger stops in first few seconds without output") | Python Debugger stops in first few seconds without output |
 |  OP:444  |[1291](/issues/1291 "Add debugger data breakpoints") | Add debugger data breakpoints |
 |  OP:384  |[1328](/issues/1328 "View thread state in debugger without breakpoints") | View thread state in debugger without breakpoints |
 |  OP:349  |[1355](/issues/1355 "Pytest debugger isn't compatible with vscode+playwright") | Pytest debugger isn't compatible with vscode+playwright |
 |  OP:327  |[1377](/issues/1377 "Show Python Process names in debugger") | Show Python Process names in debugger |
 |  OP:322  |[1382](/issues/1382 "Python debugger frozen eternall with poetry env") | Python debugger frozen eternall with poetry env |
 |  OP:321  |[1384](/issues/1384 "Cannot run Debugger on macOS (pydev debugger: critical: unable to get real case for file. Details:)") | Cannot run Debugger on macOS (pydev debugger: critical: unable to get real case for file. Details:) |
 |  OP:252  |[1463](/issues/1463 "Timeout waiting for debugger connection to running pythonnet process") | Timeout waiting for debugger connection to running pythonnet process |
 |  OP:252  |[1465](/issues/1465 "Cannot attach Python debugger to debugpy within a future inside Docker container") | Cannot attach Python debugger to debugpy within a future inside Docker container |
 |  OP:224  |[1488](/issues/1488 "Add PyQt6 and PySide6 support for python debugger to avoid import errors and fix breakpoint issue") | Add PyQt6 and PySide6 support for python debugger to avoid import errors and fix breakpoint issue |
 |  OP:189  |[1497](/issues/1497 "Debugger hits breakpoints, but doesn't retrieve variables or continue execution") | Debugger hits breakpoints, but doesn't retrieve variables or continue execution |
 |  OP:187  |[1499](/issues/1499 "Python debugger changes __file__") | Python debugger changes __file__ |
 |  OP:181  |[1505](/issues/1505 "Python Debugger not showing NumPy array values after update - NOT resolved") | Python Debugger not showing NumPy array values after update - NOT resolved |
 |  OP:181  |[1507](/issues/1507 "[sys.monitoring] API to hide threads and files from debugger") | [sys.monitoring] API to hide threads and files from debugger |
 |  OP:125  |[1548](/issues/1548 "[sys.monitoring] Local variable cleanup is delayed by debugger") | [sys.monitoring] Local variable cleanup is delayed by debugger |
 |  OP:113  |[1561](/issues/1561 "&quot;import akshare as ak&quot; blocks the debugger after I upgrade to python 3.12.2") | "import akshare as ak" blocks the debugger after I upgrade to python 3.12.2 |
 |  OP:83  |[1579](/issues/1579 "Debugger times out if shell startup is &quot;slow&quot;") | Debugger times out if shell startup is "slow" |
 |  OP:60  |[1593](/issues/1593 "python debugger ignores breakpoints?") | python debugger ignores breakpoints? |
 |  OP:59  |[1592](/issues/1592 "Microsoft Python Debugger not working in a virtual environment created with venv module on windows") | Microsoft Python Debugger not working in a virtual environment created with venv module on windows |
 |  OP:11  |[1630](/issues/1630 "Python debugger breakpoints are not working.") | Python debugger breakpoints are not working. |




### Issues with term 'debug'

| Days Ago | Issue | Title |
| --- | --- | --- |
 |  OP:546  |[1193](/issues/1193 "Debug crashes when running a Python2 subprocess") | Debug crashes when running a Python2 subprocess |
 |  OP:507  |[1236](/issues/1236 "Local and global variables are not available in list comprehensions in the debug console") | Local and global variables are not available in list comprehensions in the debug console |
 |  OP:446  |[1288](/issues/1288 "Can't debug if project resides in folder whose name contains special characters") | Can't debug if project resides in folder whose name contains special characters |
 |  OP:425  |[1304](/issues/1304 "Show a warning that we can't debug when &quot;--cov&quot; is in pytestArgs") | Show a warning that we can't debug when "--cov" is in pytestArgs |
 |  OP:421  |[1308](/issues/1308 "remote debug port stucks upon `nc -z` but `nc` works") | remote debug port stucks upon `nc -z` but `nc` works |
 |  OP:405  |[1315](/issues/1315 "Debugging Django in Docker container with VS Code doesn't engage debug inspector") | Debugging Django in Docker container with VS Code doesn't engage debug inspector |
 |  OP:404  |[1320](/issues/1320 "Adopt lifecycleManagedByParent for debug sessions") | Adopt lifecycleManagedByParent for debug sessions |
 |  OP:377  |[1332](/issues/1332 "Debug variable displays verbose info") | Debug variable displays verbose info |
 |  OP:362  |[1342](/issues/1342 "Can not debug multiprocess code with streamlit") | Can not debug multiprocess code with streamlit |
 |  OP:359  |[1346](/issues/1346 "Using rich.print() instead of print() in (evaluate in) debug console") | Using rich.print() instead of print() in (evaluate in) debug console |
 |  OP:284  |[1409](/issues/1409 "Debug console shows too much detail and too little data information ") | Debug console shows too much detail and too little data information  |
 |  OP:258  |[1419](/issues/1419 "[meta] debug server based on sys.monitoring for Python 3.12+") | [meta] debug server based on sys.monitoring for Python 3.12+ |
 |  OP:210  |[1491](/issues/1491 "debug python with docker-compose cannot hit breakpoint") | debug python with docker-compose cannot hit breakpoint |
 |  OP:199  |[1493](/issues/1493 "tries to debug a wrong file, behaves weird if file is moved") | tries to debug a wrong file, behaves weird if file is moved |
 |  OP:195  |[1494](/issues/1494 "Debug windows stuck on loading local variable(large img tensor) when debug pytorch code.") | Debug windows stuck on loading local variable(large img tensor) when debug pytorch code. |
 |  OP:185  |[1501](/issues/1501 "Allow multiple debug connections in VS code using the same port") | Allow multiple debug connections in VS code using the same port |
 |  OP:152  |[1527](/issues/1527 "Debug should skip arguments assignment for step into action by default") | Debug should skip arguments assignment for step into action by default |
 |  OP:139  |[1539](/issues/1539 "Python Debug console reports multiple wxPyDeprecationWarning when using wxPython 4.2.1") | Python Debug console reports multiple wxPyDeprecationWarning when using wxPython 4.2.1 |
 |  OP:101  |[1568](/issues/1568 "Allow clients to debug code from command line, like pdb or epdb ") | Allow clients to debug code from command line, like pdb or epdb  |
 |  OP:79  |[1582](/issues/1582 "Run remote Debug success,But restart Debug errorInfo-show: connect ECONNREFUSED ip-address:5678") | Run remote Debug success,But restart Debug errorInfo-show: connect ECONNREFUSED ip-address:5678 |
 |  OP:47  |[1599](/issues/1599 "Remotely debug Python code on Linux from Microsoft Visual Studio 2022:  cannot set breakpoint (Breakpoint in file that does not exist)") | Remotely debug Python code on Linux from Microsoft Visual Studio 2022:  cannot set breakpoint (Breakpoint in file that does not exist) |
 |  OP:34  |[1609](/issues/1609 "Troubles when expanding self object during debug in constructor") | Troubles when expanding self object during debug in constructor |
 |  OP:12  |[1627](/issues/1627 "Can't enter core python libraries in debug mode") | Can't enter core python libraries in debug mode |
 |  OP:7  |[1633](/issues/1633 "interactive plot window not showing under wayland in debug mode for matplotlib=3.9") | interactive plot window not showing under wayland in debug mode for matplotlib=3.9 |
 |  OP:0  |[1642](/issues/1642 "Unable to set new variables or update exiting variables in debug console") | Unable to set new variables or update exiting variables in debug console |




### Issues with term 'debugging'

| Days Ago | Issue | Title |
| --- | --- | --- |
 |  OP:1717  |[165](/issues/165 "Entry points aren't being found while test debugging") | Entry points aren't being found while test debugging |
 |  OP:614  |[1134](/issues/1134 "async code debugging") | async code debugging |
 |  OP:577  |[1167](/issues/1167 "Debugging support lazy variables") | Debugging support lazy variables |
 |  OP:501  |[1245](/issues/1245 "Don't collapse nested lists/tuples into ... in debugger | make debugging more like PyCharm") | Don't collapse nested lists/tuples into ... in debugger | make debugging more like PyCharm |
 |  OP:494  |[1250](/issues/1250 "Debugging code with `pexpect.spawn` is taking 5s longer  ") | Debugging code with `pexpect.spawn` is taking 5s longer   |
 |  OP:454  |[1284](/issues/1284 "Debugging under python 3.11 will skip the breakpoint") | Debugging under python 3.11 will skip the breakpoint |
 |  OP:454  |[1285](/issues/1285 "Terminal cuts off long strings when debugging") | Terminal cuts off long strings when debugging |
 |  OP:428  |[1303](/issues/1303 "When using python debugging class properties are showed after private and protected members.") | When using python debugging class properties are showed after private and protected members. |
 |  OP:414  |[1311](/issues/1311 "Support debugging console scripts from launch.json") | Support debugging console scripts from launch.json |
 |  OP:405  |[1315](/issues/1315 "Debugging Django in Docker container with VS Code doesn't engage debug inspector") | Debugging Django in Docker container with VS Code doesn't engage debug inspector |
 |  OP:368  |[1336](/issues/1336 "VSCode sometimes stuck on breakpoint when debugging an embedded interpreter") | VSCode sometimes stuck on breakpoint when debugging an embedded interpreter |
 |  OP:308  |[1400](/issues/1400 "Debugging Panel Apps written as IPython Notebook in VSCode") | Debugging Panel Apps written as IPython Notebook in VSCode |
 |  OP:226  |[1485](/issues/1485 "Bug while debugging list comprehension (01)") | Bug while debugging list comprehension (01) |
 |  OP:226  |[1486](/issues/1486 "Bug while debugging list comprehension (02)") | Bug while debugging list comprehension (02) |
 |  OP:184  |[1503](/issues/1503 "Support debugging of indirect child processes.") | Support debugging of indirect child processes. |
 |  OP:166  |[1521](/issues/1521 "Running debugging shows errors, but the functionality is normal.") | Running debugging shows errors, but the functionality is normal. |
 |  OP:154  |[1525](/issues/1525 "Display order of properties of a tensor object in debugging") | Display order of properties of a tensor object in debugging |
 |  OP:145  |[1533](/issues/1533 "Duplicate path error suddenly appears on debugging only") | Duplicate path error suddenly appears on debugging only |
 |  OP:139  |[1536](/issues/1536 "Subprocess debugging not working when --pid and --configure-subProcess are used") | Subprocess debugging not working when --pid and --configure-subProcess are used |
 |  OP:132  |[1542](/issues/1542 "Puzzling behavior of debugging with debugpy") | Puzzling behavior of debugging with debugpy |
 |  OP:104  |[1565](/issues/1565 "Skip nonrelevant `case`s in `match/case` during step by step debugging") | Skip nonrelevant `case`s in `match/case` during step by step debugging |
 |  OP:70  |[1589](/issues/1589 "[DTS] Unable to attach to embedded python in mixed mode debugging") | [DTS] Unable to attach to embedded python in mixed mode debugging |
 |  OP:30  |[1615](/issues/1615 "Got stuck when debugging a multi-process program in PyTorch") | Got stuck when debugging a multi-process program in PyTorch |




### Issues with term 'breakpoints'

| Days Ago | Issue | Title |
| --- | --- | --- |
 |  OP:1314  |[510](/issues/510 "Stop at breakpoints during evaluate request (recursive debugging)") | Stop at breakpoints during evaluate request (recursive debugging) |
 |  OP:696  |[1041](/issues/1041 "Breakpoints on secondary threads don't trigger when using PyQt5") | Breakpoints on secondary threads don't trigger when using PyQt5 |
 |  OP:444  |[1291](/issues/1291 "Add debugger data breakpoints") | Add debugger data breakpoints |
 |  OP:384  |[1328](/issues/1328 "View thread state in debugger without breakpoints") | View thread state in debugger without breakpoints |
 |  OP:257  |[1429](/issues/1429 "[sys.monitoring] Breakpoints") | [sys.monitoring] Breakpoints |
 |  OP:257  |[1434](/issues/1434 "[sys.monitoring] Snap breakpoints to valid locations") | [sys.monitoring] Snap breakpoints to valid locations |
 |  OP:256  |[1436](/issues/1436 "[sys.monitoring] Exception breakpoints") | [sys.monitoring] Exception breakpoints |
 |  OP:251  |[1466](/issues/1466 "[sys.monitoring] Django breakpoints") | [sys.monitoring] Django breakpoints |
 |  OP:251  |[1467](/issues/1467 "[sys.monitoring] Flask (Jinja) breakpoints") | [sys.monitoring] Flask (Jinja) breakpoints |
 |  OP:224  |[1489](/issues/1489 "Breakpoints not triggered in Flask app") | Breakpoints not triggered in Flask app |
 |  OP:178  |[1514](/issues/1514 "Breakpoints ignored if file extension is not `.py`") | Breakpoints ignored if file extension is not `.py` |
 |  OP:133  |[1541](/issues/1541 "Cannot set remote breakpoints in Visual Studio Community 2022 (Version 17.9.3)") | Cannot set remote breakpoints in Visual Studio Community 2022 (Version 17.9.3) |
 |  OP:11  |[1630](/issues/1630 "Python debugger breakpoints are not working.") | Python debugger breakpoints are not working. |




### Issues with term 'code'

| Days Ago | Issue | Title |
| --- | --- | --- |
 |  OP:633  |[1114](/issues/1114 "Display Python asyncio Tasks in VS Code Debugger") | Display Python asyncio Tasks in VS Code Debugger |
 |  OP:614  |[1134](/issues/1134 "async code debugging") | async code debugging |
 |  OP:494  |[1250](/issues/1250 "Debugging code with `pexpect.spawn` is taking 5s longer  ") | Debugging code with `pexpect.spawn` is taking 5s longer   |
 |  OP:405  |[1315](/issues/1315 "Debugging Django in Docker container with VS Code doesn't engage debug inspector") | Debugging Django in Docker container with VS Code doesn't engage debug inspector |
 |  OP:362  |[1342](/issues/1342 "Can not debug multiprocess code with streamlit") | Can not debug multiprocess code with streamlit |
 |  OP:299  |[1405](/issues/1405 "F10 to step through the code is slow") | F10 to step through the code is slow |
 |  OP:185  |[1501](/issues/1501 "Allow multiple debug connections in VS code using the same port") | Allow multiple debug connections in VS code using the same port |
 |  OP:181  |[1509](/issues/1509 "[sys.monitoring] Just my code") | [sys.monitoring] Just my code |
 |  OP:166  |[1522](/issues/1522 "warning on code property") | warning on code property |
 |  OP:139  |[1538](/issues/1538 "Breakpoint set in source code are not hit when attaching to a devcontainer process") | Breakpoint set in source code are not hit when attaching to a devcontainer process |
 |  OP:101  |[1568](/issues/1568 "Allow clients to debug code from command line, like pdb or epdb ") | Allow clients to debug code from command line, like pdb or epdb  |
 |  OP:76  |[1584](/issues/1584 "Attach to PID of application embedding Python not working in VS Code on Windows") | Attach to PID of application embedding Python not working in VS Code on Windows |
 |  OP:47  |[1599](/issues/1599 "Remotely debug Python code on Linux from Microsoft Visual Studio 2022:  cannot set breakpoint (Breakpoint in file that does not exist)") | Remotely debug Python code on Linux from Microsoft Visual Studio 2022:  cannot set breakpoint (Breakpoint in file that does not exist) |




### Issues with term 'breakpoint'

| Days Ago | Issue | Title |
| --- | --- | --- |
 |  OP:711  |[1101](/issues/1101 "Improve inline breakpoint experience to be similar to TypeScript's  ") | Improve inline breakpoint experience to be similar to TypeScript's   |
 |  OP:474  |[1272](/issues/1272 "When using debugger in remote environment it hangs and never successfully breaks when reaching a breakpoint") | When using debugger in remote environment it hangs and never successfully breaks when reaching a breakpoint |
 |  OP:467  |[1277](/issues/1277 "Breakpoint not working in Odoo with multiple workers (gevent)") | Breakpoint not working in Odoo with multiple workers (gevent) |
 |  OP:454  |[1284](/issues/1284 "Debugging under python 3.11 will skip the breakpoint") | Debugging under python 3.11 will skip the breakpoint |
 |  OP:368  |[1336](/issues/1336 "VSCode sometimes stuck on breakpoint when debugging an embedded interpreter") | VSCode sometimes stuck on breakpoint when debugging an embedded interpreter |
 |  OP:257  |[1433](/issues/1433 "[sys.monitoring] Breakpoint verification") | [sys.monitoring] Breakpoint verification |
 |  OP:224  |[1488](/issues/1488 "Add PyQt6 and PySide6 support for python debugger to avoid import errors and fix breakpoint issue") | Add PyQt6 and PySide6 support for python debugger to avoid import errors and fix breakpoint issue |
 |  OP:210  |[1491](/issues/1491 "debug python with docker-compose cannot hit breakpoint") | debug python with docker-compose cannot hit breakpoint |
 |  OP:194  |[1495](/issues/1495 "Hitting a breakpoint in symlinked file opens a new instance of the file") | Hitting a breakpoint in symlinked file opens a new instance of the file |
 |  OP:168  |[1519](/issues/1519 "`set_trace`-releated deadlock when running inside Ansible with explicit listen-and-debug breakpoint") | `set_trace`-releated deadlock when running inside Ansible with explicit listen-and-debug breakpoint |
 |  OP:139  |[1538](/issues/1538 "Breakpoint set in source code are not hit when attaching to a devcontainer process") | Breakpoint set in source code are not hit when attaching to a devcontainer process |
 |  OP:47  |[1599](/issues/1599 "Remotely debug Python code on Linux from Microsoft Visual Studio 2022:  cannot set breakpoint (Breakpoint in file that does not exist)") | Remotely debug Python code on Linux from Microsoft Visual Studio 2022:  cannot set breakpoint (Breakpoint in file that does not exist) |




### Issues with term 'debugpy'

| Days Ago | Issue | Title |
| --- | --- | --- |
 |  OP:439  |[1296](/issues/1296 "Debugpy unable to start Flask server") | Debugpy unable to start Flask server |
 |  OP:376  |[1333](/issues/1333 "Release debugpy source as tar.gz instead of zip") | Release debugpy source as tar.gz instead of zip |
 |  OP:367  |[1338](/issues/1338 "Debugpy Server Disconnects Unexpectedly when Inspecting Variable") | Debugpy Server Disconnects Unexpectedly when Inspecting Variable |
 |  OP:312  |[1393](/issues/1393 "Misnamed wheel file name for Debugpy 1.8.0") | Misnamed wheel file name for Debugpy 1.8.0 |
 |  OP:311  |[1396](/issues/1396 "Cannot run Debugpy - ms-python.debugpy CANNOT USE these API proposals (Macos M1)") | Cannot run Debugpy - ms-python.debugpy CANNOT USE these API proposals (Macos M1) |
 |  OP:311  |[1397](/issues/1397 "Can't use debugpy in macos/vscode. Immediately Finishes (CANNOT USE these API proposals 'portsAttributes')") | Can't use debugpy in macos/vscode. Immediately Finishes (CANNOT USE these API proposals 'portsAttributes') |
 |  OP:281  |[1410](/issues/1410 "How to make debugpy respect custom eval frame functions?") | How to make debugpy respect custom eval frame functions? |
 |  OP:252  |[1465](/issues/1465 "Cannot attach Python debugger to debugpy within a future inside Docker container") | Cannot attach Python debugger to debugpy within a future inside Docker container |
 |  OP:149  |[1529](/issues/1529 "api to retrieve debugpy endpoint and access token?") | api to retrieve debugpy endpoint and access token? |
 |  OP:132  |[1542](/issues/1542 "Puzzling behavior of debugging with debugpy") | Puzzling behavior of debugging with debugpy |
 |  OP:38  |[1607](/issues/1607 "Enable debugpy to work with domain sockets") | Enable debugpy to work with domain sockets |




### Issues with term 'attach'

| Days Ago | Issue | Title |
| --- | --- | --- |
 |  OP:1509  |[286](/issues/286 "Attach to local process assumes i386 architecture? ") | Attach to local process assumes i386 architecture?  |
 |  OP:278  |[1413](/issues/1413 "Attach function doesn't work with ubuntu 22.04") | Attach function doesn't work with ubuntu 22.04 |
 |  OP:252  |[1464](/issues/1464 "remote vscode attach process pid failed") | remote vscode attach process pid failed |
 |  OP:252  |[1465](/issues/1465 "Cannot attach Python debugger to debugpy within a future inside Docker container") | Cannot attach Python debugger to debugpy within a future inside Docker container |
 |  OP:239  |[1476](/issues/1476 "Python3.12 Unable to attach to PID (Jupyter python process)") | Python3.12 Unable to attach to PID (Jupyter python process) |
 |  OP:231  |[1480](/issues/1480 "Undocumented gdb Version (> 11.1) Requirements for Python Attach Mode ") | Undocumented gdb Version (> 11.1) Requirements for Python Attach Mode  |
 |  OP:101  |[1570](/issues/1570 "Attach to process doesn't work for python 3.12") | Attach to process doesn't work for python 3.12 |
 |  OP:76  |[1584](/issues/1584 "Attach to PID of application embedding Python not working in VS Code on Windows") | Attach to PID of application embedding Python not working in VS Code on Windows |
 |  OP:70  |[1589](/issues/1589 "[DTS] Unable to attach to embedded python in mixed mode debugging") | [DTS] Unable to attach to embedded python in mixed mode debugging |




### Issues with term 'process'

| Days Ago | Issue | Title |
| --- | --- | --- |
 |  OP:1509  |[286](/issues/286 "Attach to local process assumes i386 architecture? ") | Attach to local process assumes i386 architecture?  |
 |  OP:1063  |[712](/issues/712 "notification like &quot;Failed launch debugger for child process xxxx&quot;.") | notification like "Failed launch debugger for child process xxxx". |
 |  OP:474  |[1271](/issues/1271 "Timeout while using vscode debugger with any program which spawns a process") | Timeout while using vscode debugger with any program which spawns a process |
 |  OP:327  |[1377](/issues/1377 "Show Python Process names in debugger") | Show Python Process names in debugger |
 |  OP:252  |[1463](/issues/1463 "Timeout waiting for debugger connection to running pythonnet process") | Timeout waiting for debugger connection to running pythonnet process |
 |  OP:252  |[1464](/issues/1464 "remote vscode attach process pid failed") | remote vscode attach process pid failed |
 |  OP:139  |[1538](/issues/1538 "Breakpoint set in source code are not hit when attaching to a devcontainer process") | Breakpoint set in source code are not hit when attaching to a devcontainer process |
 |  OP:101  |[1570](/issues/1570 "Attach to process doesn't work for python 3.12") | Attach to process doesn't work for python 3.12 |




### Issues with term 'variable'

| Days Ago | Issue | Title |
| --- | --- | --- |
 |  OP:1494  |[317](/issues/317 "Make variable order for dict keys configurable") | Make variable order for dict keys configurable |
 |  OP:558  |[1179](/issues/1179 "Support DAP variable paging") | Support DAP variable paging |
 |  OP:377  |[1332](/issues/1332 "Debug variable displays verbose info") | Debug variable displays verbose info |
 |  OP:367  |[1338](/issues/1338 "Debugpy Server Disconnects Unexpectedly when Inspecting Variable") | Debugpy Server Disconnects Unexpectedly when Inspecting Variable |
 |  OP:256  |[1439](/issues/1439 "[sys.monitoring] Variable children") | [sys.monitoring] Variable children |
 |  OP:256  |[1440](/issues/1440 "[sys.monitoring] Variable values") | [sys.monitoring] Variable values |
 |  OP:125  |[1548](/issues/1548 "[sys.monitoring] Local variable cleanup is delayed by debugger") | [sys.monitoring] Local variable cleanup is delayed by debugger |




### Issues with term 'children'

| Days Ago | Issue | Title |
| --- | --- | --- |
 |  OP:256  |[1439](/issues/1439 "[sys.monitoring] Variable children") | [sys.monitoring] Variable children |
 |  OP:256  |[1456](/issues/1456 "[sys.monitoring] Children - object attributes") | [sys.monitoring] Children - object attributes |
 |  OP:256  |[1457](/issues/1457 "[sys.monitoring] Children - sequence & set items") | [sys.monitoring] Children - sequence & set items |
 |  OP:256  |[1458](/issues/1458 "[sys.monitoring] Children - mapping items") | [sys.monitoring] Children - mapping items |
 |  OP:256  |[1459](/issues/1459 "[sys.monitoring] Children - numpy collections") | [sys.monitoring] Children - numpy collections |
 |  OP:256  |[1460](/issues/1460 "[sys.monitoring] Children - pandas collections") | [sys.monitoring] Children - pandas collections |
 |  OP:256  |[1461](/issues/1461 "[sys.monitoring] Children - tensorflow collections") | [sys.monitoring] Children - tensorflow collections |




### Issues with term 'variables'

| Days Ago | Issue | Title |
| --- | --- | --- |
 |  OP:577  |[1167](/issues/1167 "Debugging support lazy variables") | Debugging support lazy variables |
 |  OP:507  |[1236](/issues/1236 "Local and global variables are not available in list comprehensions in the debug console") | Local and global variables are not available in list comprehensions in the debug console |
 |  OP:413  |[1310](/issues/1310 "Variables pane stuck loading") | Variables pane stuck loading |
 |  OP:189  |[1497](/issues/1497 "Debugger hits breakpoints, but doesn't retrieve variables or continue execution") | Debugger hits breakpoints, but doesn't retrieve variables or continue execution |
 |  OP:80  |[1581](/issues/1581 "Configure variables view presentation of `len()`") | Configure variables view presentation of `len()` |
 |  OP:0  |[1642](/issues/1642 "Unable to set new variables or update exiting variables in debug console") | Unable to set new variables or update exiting variables in debug console |




### Issues with term 'console'

| Days Ago | Issue | Title |
| --- | --- | --- |
 |  OP:507  |[1236](/issues/1236 "Local and global variables are not available in list comprehensions in the debug console") | Local and global variables are not available in list comprehensions in the debug console |
 |  OP:414  |[1311](/issues/1311 "Support debugging console scripts from launch.json") | Support debugging console scripts from launch.json |
 |  OP:359  |[1346](/issues/1346 "Using rich.print() instead of print() in (evaluate in) debug console") | Using rich.print() instead of print() in (evaluate in) debug console |
 |  OP:284  |[1409](/issues/1409 "Debug console shows too much detail and too little data information ") | Debug console shows too much detail and too little data information  |
 |  OP:139  |[1539](/issues/1539 "Python Debug console reports multiple wxPyDeprecationWarning when using wxPython 4.2.1") | Python Debug console reports multiple wxPyDeprecationWarning when using wxPython 4.2.1 |
 |  OP:0  |[1642](/issues/1642 "Unable to set new variables or update exiting variables in debug console") | Unable to set new variables or update exiting variables in debug console |




### Issues with term 'update'

| Days Ago | Issue | Title |
| --- | --- | --- |
 |  OP:1816  |[182](/issues/182 "Update CI and testing for ARM") | Update CI and testing for ARM |
 |  OP:1816  |[181](/issues/181 "Update CI and testing for ARM64") | Update CI and testing for ARM64 |
 |  OP:883  |[856](/issues/856 "update setup.py license field") | update setup.py license field |
 |  OP:181  |[1505](/issues/1505 "Python Debugger not showing NumPy array values after update - NOT resolved") | Python Debugger not showing NumPy array values after update - NOT resolved |
 |  OP:0  |[1642](/issues/1642 "Unable to set new variables or update exiting variables in debug console") | Unable to set new variables or update exiting variables in debug console |




### Issues with term 'stuck'

| Days Ago | Issue | Title |
| --- | --- | --- |
 |  OP:1629  |[43](/issues/43 "Subprocesses are stuck if they have no known parent") | Subprocesses are stuck if they have no known parent |
 |  OP:413  |[1310](/issues/1310 "Variables pane stuck loading") | Variables pane stuck loading |
 |  OP:368  |[1336](/issues/1336 "VSCode sometimes stuck on breakpoint when debugging an embedded interpreter") | VSCode sometimes stuck on breakpoint when debugging an embedded interpreter |
 |  OP:195  |[1494](/issues/1494 "Debug windows stuck on loading local variable(large img tensor) when debug pytorch code.") | Debug windows stuck on loading local variable(large img tensor) when debug pytorch code. |
 |  OP:30  |[1615](/issues/1615 "Got stuck when debugging a multi-process program in PyTorch") | Got stuck when debugging a multi-process program in PyTorch |




### Issues with term 'run'

| Days Ago | Issue | Title |
| --- | --- | --- |
 |  OP:1524  |[268](/issues/268 "Add documentation for clients that we expect python to be able to run directory") | Add documentation for clients that we expect python to be able to run directory |
 |  OP:1301  |[519](/issues/519 "Tests run on all platforms with non-latin character paths or pseudo loc") | Tests run on all platforms with non-latin character paths or pseudo loc |
 |  OP:321  |[1384](/issues/1384 "Cannot run Debugger on macOS (pydev debugger: critical: unable to get real case for file. Details:)") | Cannot run Debugger on macOS (pydev debugger: critical: unable to get real case for file. Details:) |
 |  OP:311  |[1396](/issues/1396 "Cannot run Debugpy - ms-python.debugpy CANNOT USE these API proposals (Macos M1)") | Cannot run Debugpy - ms-python.debugpy CANNOT USE these API proposals (Macos M1) |
 |  OP:79  |[1582](/issues/1582 "Run remote Debug success,But restart Debug errorInfo-show: connect ECONNREFUSED ip-address:5678") | Run remote Debug success,But restart Debug errorInfo-show: connect ECONNREFUSED ip-address:5678 |




### Issues with term 'running'

| Days Ago | Issue | Title |
| --- | --- | --- |
 |  OP:1467  |[342](/issues/342 "test_nodebug broken if not running under tox") | test_nodebug broken if not running under tox |
 |  OP:546  |[1193](/issues/1193 "Debug crashes when running a Python2 subprocess") | Debug crashes when running a Python2 subprocess |
 |  OP:252  |[1463](/issues/1463 "Timeout waiting for debugger connection to running pythonnet process") | Timeout waiting for debugger connection to running pythonnet process |
 |  OP:168  |[1519](/issues/1519 "`set_trace`-releated deadlock when running inside Ansible with explicit listen-and-debug breakpoint") | `set_trace`-releated deadlock when running inside Ansible with explicit listen-and-debug breakpoint |
 |  OP:166  |[1521](/issues/1521 "Running debugging shows errors, but the functionality is normal.") | Running debugging shows errors, but the functionality is normal. |




### Issues with term 'error'

| Days Ago | Issue | Title |
| --- | --- | --- |
 |  OP:1364  |[460](/issues/460 "Repeated debugpy.listen() calls should be an error") | Repeated debugpy.listen() calls should be an error |
 |  OP:145  |[1533](/issues/1533 "Duplicate path error suddenly appears on debugging only") | Duplicate path error suddenly appears on debugging only |
 |  OP:79  |[1583](/issues/1583 "IPython Qt incorrect string formatting in raised error") | IPython Qt incorrect string formatting in raised error |
 |  OP:70  |[1587](/issues/1587 "Error : TypeError: '>' not supported between instances of 'int' and 'NoneType'") | Error : TypeError: '>' not supported between instances of 'int' and 'NoneType' |
 |  OP:39  |[1604](/issues/1604 "Fatal Python Error on debugpy.adapter shutdown") | Fatal Python Error on debugpy.adapter shutdown |




### Issues with term 'like'

| Days Ago | Issue | Title |
| --- | --- | --- |
 |  OP:1063  |[712](/issues/712 "notification like &quot;Failed launch debugger for child process xxxx&quot;.") | notification like "Failed launch debugger for child process xxxx". |
 |  OP:501  |[1245](/issues/1245 "Don't collapse nested lists/tuples into ... in debugger | make debugging more like PyCharm") | Don't collapse nested lists/tuples into ... in debugger | make debugging more like PyCharm |
 |  OP:298  |[1406](/issues/1406 "IORedirector doesn't behave like IOBase causing problems interacting with stdout and stderr") | IORedirector doesn't behave like IOBase causing problems interacting with stdout and stderr |
 |  OP:122  |[1553](/issues/1553 "possible to effectively get something like &quot;jupyter.debugJustMyCode&quot;: 'sometimes true'?") | possible to effectively get something like "jupyter.debugJustMyCode": 'sometimes true'? |
 |  OP:101  |[1568](/issues/1568 "Allow clients to debug code from command line, like pdb or epdb ") | Allow clients to debug code from command line, like pdb or epdb  |




### Issues with term 'remote'

| Days Ago | Issue | Title |
| --- | --- | --- |
 |  OP:474  |[1272](/issues/1272 "When using debugger in remote environment it hangs and never successfully breaks when reaching a breakpoint") | When using debugger in remote environment it hangs and never successfully breaks when reaching a breakpoint |
 |  OP:421  |[1308](/issues/1308 "remote debug port stucks upon `nc -z` but `nc` works") | remote debug port stucks upon `nc -z` but `nc` works |
 |  OP:252  |[1464](/issues/1464 "remote vscode attach process pid failed") | remote vscode attach process pid failed |
 |  OP:133  |[1541](/issues/1541 "Cannot set remote breakpoints in Visual Studio Community 2022 (Version 17.9.3)") | Cannot set remote breakpoints in Visual Studio Community 2022 (Version 17.9.3) |
 |  OP:79  |[1582](/issues/1582 "Run remote Debug success,But restart Debug errorInfo-show: connect ECONNREFUSED ip-address:5678") | Run remote Debug success,But restart Debug errorInfo-show: connect ECONNREFUSED ip-address:5678 |




### Issues with term 'file'

| Days Ago | Issue | Title |
| --- | --- | --- |
 |  OP:312  |[1393](/issues/1393 "Misnamed wheel file name for Debugpy 1.8.0") | Misnamed wheel file name for Debugpy 1.8.0 |
 |  OP:199  |[1493](/issues/1493 "tries to debug a wrong file, behaves weird if file is moved") | tries to debug a wrong file, behaves weird if file is moved |
 |  OP:194  |[1495](/issues/1495 "Hitting a breakpoint in symlinked file opens a new instance of the file") | Hitting a breakpoint in symlinked file opens a new instance of the file |
 |  OP:178  |[1514](/issues/1514 "Breakpoints ignored if file extension is not `.py`") | Breakpoints ignored if file extension is not `.py` |
 |  OP:47  |[1599](/issues/1599 "Remotely debug Python code on Linux from Microsoft Visual Studio 2022:  cannot set breakpoint (Breakpoint in file that does not exist)") | Remotely debug Python code on Linux from Microsoft Visual Studio 2022:  cannot set breakpoint (Breakpoint in file that does not exist) |




### Issues with term 'stack'

| Days Ago | Issue | Title |
| --- | --- | --- |
 |  OP:257  |[1421](/issues/1421 "[sys.monitoring] Stack tracing") | [sys.monitoring] Stack tracing |
 |  OP:251  |[1468](/issues/1468 "[sys.monitoring] Django template stack frames") | [sys.monitoring] Django template stack frames |
 |  OP:251  |[1469](/issues/1469 "[sys.monitoring] Jinja template stack frames") | [sys.monitoring] Jinja template stack frames |
 |  OP:244  |[1474](/issues/1474 "Can't see libraries in stack trace even with &quot;justMyCode&quot;: false") | Can't see libraries in stack trace even with "justMyCode": false |
 |  OP:92  |[1572](/issues/1572 "The dead thread  won't disappear in call stack monitor") | The dead thread  won't disappear in call stack monitor |




### Issues with term 'safe'

| Days Ago | Issue | Title |
| --- | --- | --- |
 |  OP:256  |[1447](/issues/1447 "[sys.monitoring] safe repr - strings") | [sys.monitoring] safe repr - strings |
 |  OP:256  |[1448](/issues/1448 "[sys.monitoring] safe repr - collections") | [sys.monitoring] safe repr - collections |
 |  OP:256  |[1449](/issues/1449 "[sys.monitoring] safe repr - numpy") | [sys.monitoring] safe repr - numpy |
 |  OP:256  |[1450](/issues/1450 "[sys.monitoring] safe repr - pandas") | [sys.monitoring] safe repr - pandas |
 |  OP:256  |[1451](/issues/1451 "[sys.monitoring] safe repr - tensorflow") | [sys.monitoring] safe repr - tensorflow |




### Issues with term 'repr'

| Days Ago | Issue | Title |
| --- | --- | --- |
 |  OP:256  |[1447](/issues/1447 "[sys.monitoring] safe repr - strings") | [sys.monitoring] safe repr - strings |
 |  OP:256  |[1448](/issues/1448 "[sys.monitoring] safe repr - collections") | [sys.monitoring] safe repr - collections |
 |  OP:256  |[1449](/issues/1449 "[sys.monitoring] safe repr - numpy") | [sys.monitoring] safe repr - numpy |
 |  OP:256  |[1450](/issues/1450 "[sys.monitoring] safe repr - pandas") | [sys.monitoring] safe repr - pandas |
 |  OP:256  |[1451](/issues/1451 "[sys.monitoring] safe repr - tensorflow") | [sys.monitoring] safe repr - tensorflow |




### Issues with term 'set'

| Days Ago | Issue | Title |
| --- | --- | --- |
 |  OP:256  |[1457](/issues/1457 "[sys.monitoring] Children - sequence & set items") | [sys.monitoring] Children - sequence & set items |
 |  OP:139  |[1538](/issues/1538 "Breakpoint set in source code are not hit when attaching to a devcontainer process") | Breakpoint set in source code are not hit when attaching to a devcontainer process |
 |  OP:133  |[1541](/issues/1541 "Cannot set remote breakpoints in Visual Studio Community 2022 (Version 17.9.3)") | Cannot set remote breakpoints in Visual Studio Community 2022 (Version 17.9.3) |
 |  OP:47  |[1599](/issues/1599 "Remotely debug Python code on Linux from Microsoft Visual Studio 2022:  cannot set breakpoint (Breakpoint in file that does not exist)") | Remotely debug Python code on Linux from Microsoft Visual Studio 2022:  cannot set breakpoint (Breakpoint in file that does not exist) |
 |  OP:0  |[1642](/issues/1642 "Unable to set new variables or update exiting variables in debug console") | Unable to set new variables or update exiting variables in debug console |
